src/hip_module.cpp

```diff
@@ -587,6 +659,11 @@ hipError_t hipLaunchCooperativeKernelMultiDevice(hipLaunchParams* launchParamsLi
     free(gwsKds);
     free(kds);
 
+    hip_internal::ihipHostFree(tls, mg_sync_ptr);
+    for (int j = 0; j < numDevices; ++j) {
+        hip_internal::ihipHostFree(tls, mg_info_ptr[j]);
+    }
```

Comment:
Can these be freed here?  Don't we have to wait until all the kernels are finished before we can free these?

Reply:
there is a wait inside ihipHostFree to make sure that the kernels are done before freeing memories.